adjustPipelines.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+allDirectories() {
+	cat <(cat locations.properties | head -n 1 | sed 's/xtext-[^/]*/xtext-eclipse/') <(cat locations.properties | head -n 1 | sed 's/xtext-[^/]*/xtext-maven/') <(cat locations.properties) | cut -d "=" -f2- 
+}
+
+toDir() {
+	echo $1 | tr -d '\r'
+}
+
+name() {
+	echo $1 | sed 's/^.*\/\(xtext-[^/]*\)\/.*$/\1/'
+}
+
+branchname=${1:-master}
+
+isBranch() {
+	git show-branch "$branchname" &> /dev/null
+}
+
+isMaster() {
+	test $branchname != master && isBranch
+}
+
+changeDir() {
+	pushd $1 > /dev/null
+}
+
+dropDir() {
+	popd > /dev/null
+}
+
+allDirectories | while read -r line
+do
+	directory=$(toDir $line)
+	upstream=$(echo $directory/gradle/upstream-repositories.gradle)
+	if [[ -e $upstream ]]
+	then
+		changeDir $directory
+		if isBranch
+		then
+			echo "Redirecting maven repositories in $upstream to $branchname"
+			dropDir
+			allDirectories | while read -r repo
+			do
+				repository=$(toDir $repo)
+				if [[ $directory != $repository ]]
+				then
+					logicalname=$(name $repo)
+					if changeDir $repository
+					then
+						sed -i '' "s%'http://services.typefox.io/open-source/jenkins/job/$logicalname/job/[^/]*/lastSuccessfulBuild/artifact/build/maven-repository/'%jenkinsPipelineRepo('$logicalname')%" $upstream
+						isMaster ||	sed -i '' "s%jenkinsPipelineRepo('$logicalname')%'http://services.typefox.io/open-source/jenkins/job/$logicalname/job/$1/lastSuccessfulBuild/artifact/build/maven-repository/'%" $upstream
```

Comment:
sometimes we use lastStableBuild.
i wonder if we simply can set -PupstreamBranch=xxxx somehow

Reply:
theoretically yes, I was thinking about this, too. But that would really require all repos to be branched where the S&R allows to work with two out of six to be branched.